apps/meteor/app/lib/server/functions/syncRolePrioritiesForRoomIfRequired.ts

```diff
@@ -0,0 +1,68 @@
+import type { IRoom, IUser } from '@rocket.chat/core-typings';
+import { Subscriptions, Users, Rooms } from '@rocket.chat/models';
+
+import { calculateRoomRolePriorityFromRoles } from '../../../../server/lib/roles/syncRoomRolePriority';
+
+const READ_BATCH_SIZE = 1000;
+
+async function assignRoomRolePrioritiesFromMap(userIdAndRoomRolePrioritiesMap: Map<IUser['_id'], IUser['roomRolePriorities']>) {
+	const bulk = Users.col.initializeUnorderedBulkOp();
+
+	for await (const [userId, roomRolePriorities] of userIdAndRoomRolePrioritiesMap.entries()) {
```

Comment:
```suggestion
	for (const [userId, roomRolePriorities] of userIdAndRoomRolePrioritiesMap.entries()) {
```

Reply:
Or even better, you can https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map/forEach